scripts/devenv.py

```diff
@@ -1,272 +1,109 @@
+        epilog=(
+            "All arguments will be added to the 'conda create' command.\n\n"
+            "default arguments:\n  " + " ".join(DEFAULT_CREATE_ARGS)
+        ),
+        formatter_class=argparse.RawDescriptionHelpFormatter,
     )
+    ap_create.set_defaults(func=cmd_create)
 
-    remove_ap = sub.add_parser(
-        "remove",
-        help="remove an existing development environment",
-        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
-    )
-    remove_ap.set_defaults(func=command_remove)
-    add_global_args(remove_ap)
+    ap_remove = ap_sub.add_parser("remove", help="remove an existing environment")
+    ap_remove.set_defaults(func=cmd_remove)
 
-    args = ap.parse_args()
+    args, rest = ap.parse_known_args()
 
-    if shutil.which("conda") is None:
-        logging.error("conda is not installed")
-        return
+    if not all(map(shutil.which, EXECUTABLES)):
+        sys.exit("the following executables should exist in the system path: " + str(EXECUTABLES)[1:-1])
 
-    if run_returncode("conda", "build", "--version") != 0:
-        logging.error("conda-build is not installed; run 'conda install -n base -c conda-forge conda-build' to install")
-        return
+    if not all(map(is_git_repo, REPOSITORIES)):
+        sys.exit("the following directories should be valid git repositories: " + str(REPOSITORIES)[1:-1])
 
-    env = CondaEnv(args.name)
-    execution_time, _res = timed(args.func, args, env)
-    logging.info(f"finished {args.command} ({execution_time:.3f} seconds)")
+    logging.info("command %r started", args.cmd)
+    time_start = time.perf_counter()
 
+    try:
+        args.func(args, rest)
+    except subprocess.CalledProcessError:
+        logging.error("command %r failed", args.cmd)
+    except KeyboardInterrupt:
+        logging.error("command %r interrupted", args.cmd)
+    else:
+        logging.info("command %r completed", args.cmd)
 
-def command_create(args: argparse.Namespace, env: CondaEnv) -> None:
-    if env.exists():
-        logging.error(f"environment {env.name} already exists")
-        return
+    time_end = time.perf_counter()
+    logging.info("elapsed time: %.3f seconds", time_end - time_start)
 
-    location = args.location
-    if location is not None:
-        location = pathlib.Path(location).resolve()
-        if not location.is_dir():
-            logging.error(f"{location} is not a valid directory")
-            return
 
-    logging.info(
-        f"create environment {env.name}"
-        f" with packages [{', '.join(args.package)}]"
-        f" from channels [{', '.join(args.channel)}]"
-    )
-    env.create(packages=args.package, channels=args.channel)
+def cmd_create(args, rest):
+    if not rest:
+        rest = list(DEFAULT_CREATE_ARGS)
+
+    chan_args = ["--override-channels", "--strict-channel-priority"]
+    run(["conda", "install", "--yes", "--name", "base"] + chan_args + ["--channel", "conda-forge", "conda-build"])
```

Comment:
this first run modifies the base environment of users - should they be made aware? Or alternatively install conda-build in ilastik environment?

Reply:
This is a good idea, although I wonder if this is a good opportunity to make `pip install -e .` work since [using `conda develop` is not recommended](https://github.com/conda/conda-build/issues/2633#issuecomment-359572215).

Reply:
Since we have no setup.py [all conda develop does is add the directory to `conda.pth`](https://github.com/conda/conda-build/blob/cd761345ee4542eb19b989a1f44ff9cea457bdc2/conda_build/develop.py#L175) in site-packages. I don't think there's any pressure to change now. There isn't any deprecation notice for conda-develop.